crates/goose/src/systems/non_developer.rs

```diff
@@ -0,0 +1,971 @@
+                    },
+                )?;
+
+                script_path.display().to_string()
+            }
+            "applescript" => {
+                if std::env::consts::OS != "macos" {
+                    return Err(AgentError::ExecutionError(
+                        "AppleScript is only supported on macOS".into(),
+                    ));
+                }
+
+                let script_path = script_dir.path().join("script.scpt");
+                fs::write(&script_path, script).map_err(|e| {
+                    AgentError::ExecutionError(format!("Failed to write script: {}", e))
+                })?;
+
+                format!("osascript {}", script_path.display())
+            }
+            "ruby" => {
+                let script_path = script_dir.path().join("script.rb");
+                fs::write(&script_path, script).map_err(|e| {
+                    AgentError::ExecutionError(format!("Failed to write script: {}", e))
+                })?;
+
+                format!("ruby {}", script_path.display())
+            }
+            _ => unreachable!(), // Prevented by enum in tool definition
+        };
+
+        // Run the script
+        let output = Command::new("bash")
+            .arg("-c")
+            .arg(&command)
+            .output()
+            .await
+            .map_err(|e| AgentError::ExecutionError(format!("Failed to run script: {}", e)))?;
+
+        let output_str = String::from_utf8_lossy(&output.stdout).into_owned();
+        let error_str = String::from_utf8_lossy(&output.stderr).into_owned();
+
+        let mut result = if output.status.success() {
+            format!("Script completed successfully.\n\nOutput:\n{}", output_str)
+        } else {
+            format!(
+                "Script failed with error code {}.\n\nError:\n{}\nOutput:\n{}",
+                output.status, error_str, output_str
+            )
+        };
+
+        // Save output if requested
+        if save_output && !output_str.is_empty() {
+            let cache_path = self
+                .save_to_cache(output_str.as_bytes(), "script_output", "txt")
+                .await?;
+            result.push_str(&format!("\n\nOutput saved to: {}", cache_path.display()));
+
+            // Register as a resource
+            let uri = Url::from_file_path(&cache_path)
+                .map_err(|_| AgentError::ExecutionError("Invalid cache path".into()))?
+                .to_string();
+
+            let resource = Resource::new(
+                uri.clone(),
+                Some("text".to_string()),
+                Some(cache_path.to_string_lossy().into_owned()),
+            )
+            .map_err(|e| AgentError::ExecutionError(e.to_string()))?;
+
+            self.active_resources.lock().unwrap().insert(uri, resource);
+        }
+
+        Ok(vec![Content::text(result)])
+    }
+
+    // Implement cache tool functionality
+    async fn cache(&self, params: Value) -> AgentResult<Vec<Content>> {
+        let operation = params
```

Comment:
nit: rename to `command` to be consistent with the text editor tool commands in developer system

Reply:
@salman1993 rename operation to command?

Reply:
ah I see, good catch